lib/player.rb

```diff
@@ -8,14 +8,31 @@ def initialize(mark, console)
     @console = console
   end
 
-  def set_current_move
-    move_given = @console.get_input
-    if move_given.between?(1,9)
-      @current_move = move_given
-    else
-      @console.show("Please enter a valid move (1-9)")
-      set_current_move
+  def set_current_move(board)
+    move = @console.get_input
+    valid_move(board, move)
+  end
+
+  private
+
+  def valid_move(board, move)
+    if empty(board, move) && within_board(move)
+      @current_move = move
+    elsif !empty(board, move)
+      @console.show("Please choose an empty position\n")
```

Comment:
I'd move these strings to the console class. Give it methods like `@console.display_empty_move_prompt`. Then when you want to change the wording, you only have to go to the display class, which makes more sense than coming to the player class.